### PR TITLE
[wallet/desktop] feat: upload-qr component unit test

### PR DESCRIPTION
**Pull request by AnthonyLaw**
## What was the issue?
- missing unit test in the `UploadQRCode` component.

## What's the fix?
- Added unit test  `UploadQRCode` component.

### CODE REVIEW

File: __tests__/components/QRCode/UploadQRCode.spec.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+            // Assert:
+            expect(vm.invalidType).toBe(false);
+            expect(vm.imageFileName).toBe('invoice_qr.png');
+            expect(vm.$refs.qrcodeCapture.onChangeInput).toHaveBeenCalledWith({
+                target: {
+                    files: [file],
+                },
+            });
+            expect(fileReaderSpy).toHaveBeenCalledWith(file);
```
Comment (Jaguar0625) [inline __tests__/components/QRCode/UploadQRCode.spec.ts:180]: check vm.image and return code

File: setupTests.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+console.log('Setting up global test stubs...');
+
+URL.createObjectURL = jest.fn();
```
Comment (Jaguar0625) [inline setupTests.js:3]: why do you need this?

Comment (AnthonyLaw) [reply setupTests.js:3]: Initially rendering the component prompts the error `URL.createObjectURL is not a function`.
I believe some dependency is required `createObjectURL` method. So I just mock it into global jest.

File: __tests__/components/QRCode/UploadQRCode.spec.ts (hunk 1 above)
Comment (AnthonyLaw) [reply __tests__/components/QRCode/UploadQRCode.spec.ts:180]: that is an interesting part, the `fileReader.onload` is not triggered in my unit test.
I will check on this problem.

File: setupTests.js (hunk 2 above)
Comment (Jaguar0625) [reply setupTests.js:3]: why global instead of just for the specific test(s) ?

Comment (AnthonyLaw) [reply setupTests.js:3]: I did try to move the specific test, but it won't work. 

I found some good explanations here.
https://github.com/mapbox/mapbox-gl-js/issues/9889#issuecomment-664633443

**Review by Jaguar0625 — APPROVED** [review]

**Pull request merged**